lib/buildFile.js

```diff
@@ -42,8 +45,49 @@ function buildFile(destination, format, platform, dictionary, filter) {
   if (!fs.existsSync(dirname))
     fs.mkdirsSync(dirname);
 
-  fs.writeFileSync(destination, format(filterProperties(dictionary, filter), platform));
-  console.log(chalk.bold.green('✔︎') + ' ' + destination);
+  let filteredProperties = filterProperties(dictionary, filter);
+
+  // Check for property name Collisions
+  var nameCollisionObj = {};
+  filteredProperties.allProperties && filteredProperties.allProperties.forEach((propertyData) => {
+    let propertyName = propertyData.name;
+    if(!nameCollisionObj[propertyName]) {
+      nameCollisionObj[propertyName] = [];
+    }
+    nameCollisionObj[propertyName].push(propertyData);
+  });
+
+  var PROPERTY_NAME_COLLISION_WARNINGS = GroupMessages.GROUP.PropertyNameCollisionWarnings;
+  var WARNING_COLOR = 'darkorange';
+  var HIGHLIGHT_COLOR = 'orangered';
+  var HELP_COLOR = 'orange';
+  Object.keys(nameCollisionObj).map((propertyName) => {
+    if(nameCollisionObj[propertyName].length > 1) {
+      let collisions = nameCollisionObj[propertyName].map(properties => chalk.keyword(HIGHLIGHT_COLOR)(properties.path.join('.')) + '   ' + chalk.keyword(WARNING_COLOR)(properties.value)).join('\n        ');
+      GroupMessages.add(
+        PROPERTY_NAME_COLLISION_WARNINGS,
+        `Output name ${chalk.keyword(HIGHLIGHT_COLOR).bold(propertyName)} was generated by:\n        ${collisions}`
+      );
+    }
+  });
+
+  let propertyNamesCollisionCount = GroupMessages.count(PROPERTY_NAME_COLLISION_WARNINGS);
+
+  fs.writeFileSync(destination, format(filteredProperties, platform));
+  console.log(chalk.bold.green((propertyNamesCollisionCount > 0) ? '⚠️ ' : '✔︎ ') + ' ' + destination);
+
+  if(propertyNamesCollisionCount > 0) {
+    var propertyNamesCollisionWarnings = GroupMessages.flush(PROPERTY_NAME_COLLISION_WARNINGS).join('\n    ');
```

Comment:
Nit: you are using `let` in other places but `var` here, should we be consistent?

Reply:
We should switch everything to ES6 as per #153 . I'm trying to stay aligned with the technique used in each file.

Reply:
Crap, nm - using both in this file. I was moving things around a bit per our conversation and didn't change after a copy and paste. Will fix.

Reply:
Ehhhhhhh.... so I just set all root level functional variables to use var and the block scoped variable to use let. Whatcha think?

Reply:
I guess that is probably fine for now, and tackle it all in #153